pkg/agent/wireguard/client_test.go

```diff
@@ -378,3 +380,27 @@ func Test_DeletePeer(t *testing.T) {
 	})
 
 }
+
+func Test_New(t *testing.T) {
+	client := fake.NewSimpleClientset()
+	_, err := New(client, &config.NodeConfig{Name: "test"}, &config.WireGuardConfig{})
+	require.NoError(t, err)
+}
+
+func Test_Init(t *testing.T) {
+	client := getFakeClient()
+	client.linkAdd = func(link netlink.Link) error {
+		return nil
+	}
+	client.linkSetUp = func(link netlink.Link) error {
+		// Pass an invalid index to fail the test.
+		link.Attrs().Index = 0
+		return nil
+	}
+	client.gatewayConfig = &config.GatewayConfig{
+		IPv4: net.ParseIP("192.168.0.2"),
+		IPv6: net.ParseIP("fd12:ab:34:a001::11"),
+	}
+	err := client.Init()
+	assert.Equal(t, "invalid argument", err.Error())
```

Comment:
doesn't it only cover the first few lines? and it doesn't even cover the most common path that will be executed.

Reply:
Yes, I thought there will be too many changes on the original codes at the beginning. I will check and do refactor.

Reply:
I see there is a fakeWireGuardClient so it should be easy to test the remaining code